src/libsyntax/visit.rs

```diff
@@ -198,6 +198,7 @@ pub fn walk_lifetime<V: Visitor>(visitor: &mut V, lifetime: &Lifetime) {
 pub fn walk_lifetime_def<V: Visitor>(visitor: &mut V, lifetime_def: &LifetimeDef) {
     visitor.visit_lifetime(&lifetime_def.lifetime);
     walk_list!(visitor, visit_lifetime, &lifetime_def.bounds);
+    walk_list!(visitor, visit_attribute, &*lifetime_def.attrs);
```

Comment:
Its necessary (here and below) because `&bounds` is a `&Vec` while `&attrs` is a `&ThinVec`.

Right now, you cannot iterate over a `&ThinVec` directly; you need to deref the `ThinVec` to turn it into a `[T]` slice that you can then iterate over.